src/topology/uniform_space/uniform_convergence_topology.lean

```diff
@@ -588,6 +590,155 @@ begin
+    { refine ⟨compl ⁻¹' Y, preimage_mono hY𝔖, hYf.preimage (compl_injective.inj_on _), _⟩,
+      rwa [compl_subset_comm, compl_sUnion, ← preimage_compl_eq_image_compl] at hxY },
+    { refine ⟨compl '' Y, image_subset_iff.mpr hY𝔖, hYf.image _, _⟩,
+      rwa [← compl_sInter, compl_subset_compl] } }
+end
+
+/-- Key part of proving `uniform_on_fun_filter_generate`: given any uniform structure `u` on
+    `α → β`, the set of all `S : set α` for which the restriction map from `(α → β, u)` → `S →ᵤ β`
+    is uniformly continuous is a *noncovering* bornology on `α`.
+    As usual, we take complements to express that in terms of filters. -/
+def co_converging_sets (u : uniform_space (α → β)) :
+  filter α :=
+{ sets := {S | @uniform_continuous (α → β) ((Sᶜ : set α) →ᵤ β) u _ (restrict Sᶜ)},
+  univ_sets :=
+  begin
+    refine uniform_continuous_of_const (λ u v, _),
+    rw compl_univ,
+    exact funext (λ x, x.2.elim)
+  end,
+  inter_sets :=
+  begin
+    rintro s₁ s₂ (h₁ : tendsto _ _ _) (h₂ : tendsto _ _ _),
+    change tendsto _ _ _,
+    rw (uniform_fun.has_basis_uniformity _ β).tendsto_right_iff at ⊢ h₁ h₂,
+    intros U hU,
+    filter_upwards [h₁ U hU, h₂ U hU],
+    rintros ⟨u, v⟩ hu hv ⟨x, hx⟩,
+    rcases not_and_distrib.mp hx with (hx|hx),
+    { exact hu ⟨x, hx⟩ },
+    { exact hv ⟨x, hx⟩ }
+  end,
+  sets_of_superset :=
+  begin
+    rintro s₁ s₂ (h₁ : tendsto _ _ _) h₁₂,
+    change tendsto _ _ _,
+    rw (uniform_fun.has_basis_uniformity _ β).tendsto_right_iff at ⊢ h₁,
+    intros U hU,
+    have h₂₁ : s₂ᶜ ⊆ s₁ᶜ := compl_subset_compl.mpr h₁₂,
+    filter_upwards [h₁ U hU] using λ uv huv ⟨x, hx⟩, huv ⟨x, h₂₁ hx⟩
+  end }
+
+/-- Replacing `𝔖` by the *noncovering* bornology generated by `𝔖` doesn't change the uniform
+structure of `𝔖`-convergence. -/
+lemma _root_.uniform_on_fun_filter_generate :
+  𝒱(α, β, {s | sᶜ ∈ filter.generate (compl ⁻¹' 𝔖)}, _) =
+  𝒱(α, β, 𝔖, _) :=
+begin
+  refine le_antisymm (infi_le_infi_of_subset $ λ s hs, filter.generate_sets.basic $ by simpa) _,
+  calc 𝒱(α, β, 𝔖, _)
+      ≤ 𝒱(α, β, 𝔖, _) : le_rfl
+  ... ≤ 𝒱(α, β, {s | sᶜ ∈ co_converging_sets 𝒱(α, β, 𝔖, _)}, _) :
+    le_infi (λ s, le_infi $ λ hs, by rwa [← uniform_continuous_iff, ← compl_compl s])
+  ... ≤ 𝒱(α, β, {s | sᶜ ∈ filter.generate (compl ⁻¹' 𝔖)}, _) :
+    infi_le_infi_of_subset (λ s hs, _),
+  suffices : co_converging_sets 𝒱(α, β, 𝔖, _) ≤ generate (compl ⁻¹' 𝔖),
+  { exact this hs },
+  exact sets_iff_generate.mpr (λ t ht, uniform_on_fun.uniform_continuous_restrict _ _ _ ht)
+end
+
+/-- If `𝔖` and `𝔗` generate the same *noncovering* bornology (in the sense of
+    `uniform_on_fun.generate_same`), then the uniform structures of `𝔖`-convergence and
+    `𝔗`-convergence are equal. -/
+lemma generate_same.uniform_space_eq {𝔖 𝔗 : set (set α)} (H : generate_same 𝔖 𝔗) :
+  𝒱(α, β, 𝔖, _) = 𝒱(α, β, 𝔗, _) :=
+by rw [← @uniform_on_fun_filter_generate _ _ _ 𝔖, ← @uniform_on_fun_filter_generate _ _ _ 𝔗,
+        (show _ = _, from H)]
```

Comment:
That line hurts. We could add
```lean
lemma generate_same.eq {𝔖 𝔗 : set (set α)} (h : generate_same 𝔖 𝔗) :
  generate (compl ⁻¹' 𝔖) = generate (compl ⁻¹' 𝔗) := h
```
to replace `(show _ = _, from H)` by `H.eq` but I guess this isn't worth the trouble if this issue arises only once (and this definition is really a technical device anyway). Also I think the parentheses aren't doing anything, except maybe making it slightly more readable.